software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -20,49 +20,73 @@
 
 import java.util.Collection;
 
+import brooklyn.enricher.Enrichers;
 import brooklyn.entity.Entity;
 import brooklyn.entity.basic.AbstractEntity;
 import brooklyn.entity.basic.Entities;
 import brooklyn.entity.group.DynamicCluster;
 import brooklyn.entity.proxying.EntitySpec;
 import brooklyn.entity.trait.Startable;
+import brooklyn.event.AttributeSensor;
+import brooklyn.event.basic.Sensors;
 import brooklyn.location.Location;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
 public class RedisClusterImpl extends AbstractEntity implements RedisCluster {
 
-    protected RedisStore master;
-    protected DynamicCluster slaves;
+    private AttributeSensor<RedisStore> MASTER = Sensors.newSensor(RedisStore.class, "redis.master");
+    private AttributeSensor<DynamicCluster> SLAVES = Sensors.newSensor(DynamicCluster.class, "redis.slaves");
 
     public RedisClusterImpl() {
     }
 
     @Override
     public RedisStore getMaster() {
-        return master;
+        return getAttribute(MASTER);
     }
     
     @Override
     public DynamicCluster getSlaves() {
-        return slaves;
+        return getAttribute(SLAVES);
     }
     
     @Override
     public void start(Collection<? extends Location> locations) {
```

Comment:
If you go down that road of setting starting/running, then I'd probably also separate it out into the two methods - delegate to something like `doStart()` to actually invoke start on the entities.

I think you're free to instantiate the `slave` cluster before calling master.start. That will mean the web-console will show the master + slaves in the tree immediately, rather than waiting for the master to start before the cluster appears.

You could also move the instantiation of the master + slave entities into init. Then you wouldn't need to guard for null either (and don't need to call `Entities.manage()` when inside init - on completion of init, any child entities will immediately get managed).